nginx-controller/nginx/nevermerger.go

```diff
@@ -0,0 +1,43 @@
+package nginx
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+type NeverMerger struct {
```

Comment:
Ok so you want to check for duplicated hostnames in the generated Servers in any case?
My understanding was that the output of the `Mergers Merge()` method cannot contain hostname/server name collisions anymore, because that is the whole purpose of this component.

Adding a additional collision check afterwards will only be redundant self validation, which should be put into the unittests of the Merger implementation.